### PR TITLE
misc: Split up tests in daily-tests.yaml

### PR DESCRIPTION
**Pull request by mkjost0**
This splits up the gem5 library example tests by Suite UID, as right now running them together uses the runner for a long period of time.  It is important to note that doing this means additional tests from this directory will need to be manually added, such as the kvm tests.

Change-Id: Ib2a0aca08f9b51b60e9dd0528324372cf2d98c05

### CODE REVIEW

File: .github/workflows/daily-tests.yaml
Comment (powerjg) [inline .github/workflows/daily-tests.yaml]: Does this say "gem5_library_example_tests_x86_ubuntu_run_ALL_x86_64_opt" somewhere in the filename?

Comment (mkjost0) [reply .github/workflows/daily-tests.yaml]: Yeah, ${{github.job}} does that, so we don't have duplicate names in the artifact output.

**Comment by powerjg** [comment]
Let's clean up this file by using matrix before merging.

File: .github/workflows/daily-tests.yaml
Comment (BobbyRBruce) [inline .github/workflows/daily-tests.yaml]: I don't like how brittle this is, as if we rename any of these directories or even the test, things will break, but I can't see a better way to do it, so I won't block on this. It's just something to keep in mind.

File: .github/workflows/daily-tests.yaml
Comment (BobbyRBruce) [inline .github/workflows/daily-tests.yaml]: You explained to be earlier that you separated this out to it's own separate build stop because "NULL" is keyword. Are you sure you can't just put this as a string in the `image` matrix? E.g., `image: ["NULL", "ALL"...`.

File: .github/workflows/daily-tests.yaml
Comment (powerjg) [inline .github/workflows/daily-tests.yaml:24]: Can you add a comment on why `"NULL"` needs quotes?

File: .github/workflows/daily-tests.yaml
Comment (powerjg) [inline .github/workflows/daily-tests.yaml:28]: A comment here would be helpful. I think I get it, but someone new to this file may be confused.

File: .github/workflows/daily-tests.yaml
Comment (powerjg) [inline .github/workflows/daily-tests.yaml]: This is not both debug and fast

File: .github/workflows/daily-tests.yaml
Comment (powerjg) [inline .github/workflows/daily-tests.yaml]: Is this 24 hours *each* or total?

File: .github/workflows/daily-tests.yaml
Comment (powerjg) [inline .github/workflows/daily-tests.yaml:88]: Do you need to download all of them for all tests? If so, can you explain why in a comment?

**Review by powerjg — COMMENTED** [review]
Generally, this looks good. It's a bit confusing, so I think it could use more comments before merging.

File: .github/workflows/daily-tests.yaml
Comment (mkjost0) [reply .github/workflows/daily-tests.yaml:88]: This one I'll explain here as well, but because we're now using a matrix, none of the long tests can start until all of the versions of gem5 within the matrix have been built.  I was toying around with the idea of still only downloading the necessary artifacts for each test, but the ways I was thinking of to do that would make this file a lot less readable than if I were to just download everything, plus it doesn't take significantly longer to wait for everything else to download.  There were also marketplace actions to download multiple artifacts in one step (which I think using would allow me to only download necessary artifacts but in a readable way), but they weren't from GitHub so I opted not to use those either

**Comment by BobbyRBruce** [comment]
I think you've accidently included [cpu-kvm: Make using perf when using KVM CPU optional](https://github.com/gem5/gem5/pull/105/commits/188d3390d7312a1ae399a7661c9262f8958de29f) into this PR. I'd do `git rebase -i HEAD~5` and remove this change then force push with `git push -f`.

**Review by BobbyRBruce — APPROVED** [review]
I think this is fine. I don't like the 'name-artifacts' job but I accept this may be the easiest way to do it.

**Pull request merged**
